table/block_based/block_based_table_reader.cc

```diff
@@ -1470,6 +1473,43 @@ IndexBlockIter* BlockBasedTable::InitBlockIterator<IndexBlockIter>(
       block_contents_pinned, rep->user_defined_timestamps_persisted);
 }
 
+template <typename TBlocklike>
+bool BlockBasedTable::LookupAndPinBlocksInCache(
+    const BlockHandle& handle,
+    CachableEntry<TBlocklike>* out_parsed_block) const {
+  BlockCacheInterface<TBlocklike> block_cache{
+      rep_->table_options.block_cache.get()};
+
+  assert(block_cache);
+
+  // Do the lookup.
+  CacheKey key_data = GetCacheKey(rep_->base_cache_key, handle);
+  const Slice key = key_data.AsSlice();
+
+  Statistics* statistics = rep_->ioptions.statistics.get();
+
+  auto cache_handle = block_cache.LookupFull(
+      key, &rep_->create_context, GetCachePriority<TBlocklike>(), statistics,
```

Comment:
There's a conflict with #11812 . Need to create a local copy of ```rep_->create_context and add the pointer to the compression dictionary, if any. You can look at ```NewDataBlockIterator``` for reference on how to retrieve the compression dict.